### DISCUSSION TITLE
Easy way to subscribe a specified property changed

### DISCUSSION THREAD

**Discussion by GF-Huang** [canton7/Stylet#197] [Ideas]
Maybe we need something like below, rather than the `INotifyPropertyChanged.PropertyChanged` with `switch (e.PropertyName) { ... }`.

```cs
public class ViewModel : PropertyChangedBase {
    public int MyProperty { get; set; } // PropertyChanged.Fody auto implements it

    public ViewModel() {
        this.SubscribePropertyChanged(nameof(MyProperty), MyPropertyChanged);
    }
    
    private void MyPropertyChanged(object sender, XXXEventArg e) {
        Console.WriteLine($"New: {e.NewValue}, Old: {e.OldValue}.");
    }
}
```

**Comment by canton7** [comment]
That already exists:

```cs
Bind(s => s.MyProperty, MyPropertyChanged);
```

**Reply by GF-Huang** [reply]
Oh my god.. Forget me..

**Reply by GF-Huang** [reply]
Why I can call it without `this`?

![image](https://user-images.githubusercontent.com/4510984/105576509-1d2b4600-5dae-11eb-9cdd-cf5dacc7e20f.png)
![image](https://user-images.githubusercontent.com/4510984/105576513-22889080-5dae-11eb-8fdf-ca7e4a132a98.png)

**Reply by canton7** [reply]
I forgot about that weirdness. It's an extension method on INotifyPropertyChanged

**Reply by GF-Huang** [reply]
@canton7 How can I get the Old Value?

**Reply by canton7** [reply]
You don't. Sorry. It's a limitation of INotifyPropertyChanged.
